bindings/wasm/examples/src/0_basic/6_create_vp.ts

```diff
@@ -0,0 +1,179 @@
+
+    // ===========================================================================
+    // Step 7: Verifier receives the Verifiable Presentation and verifies it.
+    // ===========================================================================
+
+    // Deserialize the presentation from the holder.
+    const presentation = Presentation.fromJSON(signedVpJSON);
+
+    // The verifier wants the following requirements to be satisfied:
+    // - Signature verification (including checking the requested challenge to mitigate replay attacks)
+    // - Presentation validation must fail if credentials expiring within the next 10 hours are encountered
+    // - The presentation holder must always be the subject, regardless of the presence of the nonTransferable property
+    // - The issuance date must not be in the future.
+
+    // Declare that the challenge must match our expectation:
+    const presentationVerifierOptions = new VerifierOptions({
+        challenge: "475a7984-1bb5-4c4c-a56f-822bccd46440",
+        allowExpired: false,
+    });
+
+    // Declare that any credential contained in the presentation are not allowed to expire within the next 10 hours:
+    const earliestExpiryDate = Timestamp.nowUTC().checkedAdd(Duration.hours(10));
+    const credentialValidationOptions = new CredentialValidationOptions({
+        earliestExpiryDate: earliestExpiryDate,
+    });
+
+    // Declare that the presentation holder's DID must match the subject ID on all credentials in the presentation.
+    const subjectHolderRelationship = SubjectHolderRelationship.AlwaysSubject;
+
+    const presentationValidationOptions = new PresentationValidationOptions({
+        sharedValidationOptions: credentialValidationOptions,
+        presentationVerifierOptions: presentationVerifierOptions,
+        subjectHolderRelationship: subjectHolderRelationship,
+    });
+
+    // In order to validate presentations and credentials one needs to resolve the DID Documents of
+    // the presentation holder and of credential issuers. This is something the `Resolver` can help with.
+    const resolver = new Resolver({
+        client: didClient,
+    });
+
+    // Validate the presentation and all the credentials included in it according to the validation options
+    await resolver.verifyPresentation(
+        presentation,
+        presentationValidationOptions,
+        FailFast.FirstError,
+        undefined,
+        undefined,
+    );
+
+    // Since no errors were thrown by `verifyPresentation` we know that the validation was successful.
+    console.log(`VP successfully validated`);
+
+    // Note that the `verifyPresentation` method we called automatically resolved all DID Documents that are necessary to validate the presentation.
+    // It is also possible to supply extra arguments to avoid some resolutions if one already has up-to-date resolved documents of
+    // either the holder or issuers (see the method's documentation).
```

Comment:
I think this should be moved to above the line where `verifyPresentation` is called. Otherwise, it kind of sits in the void down here.